### ISSUE TITLE
Working on Extension - Publish in own repo or add to yours?

### ISSUE THREAD

**Issue by lambricm** [alexwlchan/ao3#5]
I was working on an extension to your API for personal reasons and wanted to know if you wanted to add it to the main API or if I should just create a repo of my own and add it as an extension. I've added a way to:

* Gather chapter information - author notes, chapter summary, chapter text, number, & title
* Perform a Search given different inputs - I can handle regular search pages (similar to filtering), as well as pages where the 'tag' isn't a main tag (if you search a tag like "AAAAA" you will get this page).
* Gather a collection of works and iterate through them - I can extract work numbers via a separate class I made - called a worklist - that allows me to get all of the works in a given search. Naturally, larger searches require a _lot_ of time to come through (I was testing on 13,000 results. Searches with 14-40 results came through in a reasonable amount of time). 

I have tested it enough to get it to work in a few circumstances (not going to say extensive testing, given I am just one person and I didn't write  test cases. I just did some manual tests with searches of interest to me). I have at least used all of my functions to collect the wanted data.

As I said, this is mostly for personal use so if you don't want my code, feel free to refuse. The only part I haven't gotten working is I can't get the login part to work (actually I can't get it to work at all? Even just viewing my user data. Maybe AO3 has done some sort of update since the original login code was added), so not all the fics are accessible upon searching - only the ones that are public. Some require a user be logged-in to view them.

**Comment by alexwlchan** [comment]
As you can probably tell, I don't work on this code very much these days!

The code I have already isn't extensively tested either, so that's not an issue for me.

I have no strong opinions on where your code lives. I think it may be more visible in the main repository, but happy for you to work in a fork if you'd prefer. If you send me PRs, I'd be happy to review them and merge them, it just wouldn't be particularly quick.

**Comment by lambricm** [comment]
Sorry it took so long. I was like "woo! I can add it" and then got busy with school/finding a job that I kind of forgot. I forked it [here](https://github.com/lambricm/ao3) and have added some classes & an example file in the examples folder.

Feel free to merge if you want! And I'm not in a rush or anything. I actually made most of my additions for myself but though it'd be a good idea to share so merge it as slowly/leisurely as you'd like.
